src/sardana/taurus/qt/qtgui/extra_sardana/qtspock.py

```diff
@@ -273,7 +275,14 @@ def _set_macro_server_name(self, door):
             self._macro_server_alias = None
 
     def _set_prompts(self):
-        var = "get_ipython().config.Spock.door_alias"
+        # If traitlets >= 5.0.0 then DeferredConfigString is used for values
+        # that are not listed in the configurable classes. Get its value.
+        if self._config_passed_as_extra_arguments:
+            self.kernel_client.execute(
+                "from sardana.spock.config import Spock", silent=True)
+            var = "get_ipython().config.Spock.door_alias.get_value(Spock.door_alias)"  # noqa
+        else:
+            var = "get_ipython().config.Spock.door_alias"
```

Comment:
I didn't have a deeper look yet, but a quick test indicates that this simpler change might work as well
```suggestion
            var = "str(get_ipython().config.Spock.door_alias)"
```
Would there be a problem with this approach?

Reply:
Thanks for your comments!

I assume you want to use the simple cast to `str` it unconditionally i.e. when both, passing configuration as Taurus model and in Spock's profile. In this case it fails in the following tests:

```
FAILED workspace/sardana/src/sardana/taurus/qt/qtgui/extra_sardana/test/test_qtspock.py::MissingProfileTestCase::test_ipython_prompt - AssertionError: False is not true
FAILED workspace/sardana/src/sardana/taurus/qt/qtgui/extra_sardana/test/test_qtspock.py::CorrectProfileAfterRestartTestCase::test_ipython_prompt - AssertionError: False is not true
FAILED workspace/sardana/src/sardana/taurus/qt/qtgui/extra_sardana/test/test_qtspock.py::ProfileErrorAfterRestartTestCase::test_ipython_prompt - AssertionError: False is not true
FAILED workspace/sardana/src/sardana/taurus/qt/qtgui/extra_sardana/test/test_qtspock.py::QtSpockNoModelTestCase::test_ipython_prompt - AssertionError: False is not true
FAILED workspace/sardana/src/sardana/taurus/qt/qtgui/extra_sardana/test/test_qtspock.py::QtSpockNoModelAfterRestartTestCase::test_ipython_prompt - AssertionError: False is not true
FAILED workspace/sardana/src/sardana/taurus/qt/qtgui/extra_sardana/test/test_qtspock.py::QtSpockModelMissingProfileTestCase::test_ipython_prompt - AssertionError: False is not true
```

I debugged only the test_qtspock.py::MissingProfileTestCase::test_ipython_prompt and this produces `output_bytes='\'<LazyConfigValue {}>\''` instead of: `output_bytes='<LazyConfigValue {}>'` (in `QtSpockWidget._set_prompts_callback()`)

When I tried it conditionally, only for `self._config_passed_as_extra_arguments=True` then there are still some tests which fail.

Reply:
Sorry for not running the tests locally. When I try, the first error I get is
```
Could not create HW object: device door/demo1/1 not defined in the database !
```
I did run `sar_demo` in spock without any error message but I cannot find the door in jive. Any idea?

Reply:
As you already discovered the QtSpock tests require the [`sar_demo` test environment](https://sardana-controls.org/devel/howto_test/test_run_commands.html#sar-demo-test-environment). 

I just tried it reusing the[ reszelaz/sardana-test](https://hub.docker.com/r/reszelaz/sardana-test/) Docker image (as we do in the CI process) and this worked for me:

```
docker run -d --name=sardana-test -h sardana-test -v /homelocal/zreszela/workspace/sardana:/sardana reszelaz/sardana-test
docker exec sardana-test bash -c "cd /sardana && python3 setup.py develop"
docker exec sardana-test supervisorctl start Pool
docker exec sardana-test supervisorctl start MacroServer
docker exec sardana-test bash -c "xvfb-run -s '-screen 0 1920x1080x24' /bin/bash -c 'pytest /sardana/src/sardana/taurus/qt/qtgui/extra_sardana/test/test_qtspock.py'"
```
Note that `/homelocal/zreszela/workspace/sardana` is the path to my local Sardana Git clone.

Don't hesitate to ask if you still find problems, unfortunately the whole Sardana testing thing is not user-friendly.